src/dispatch/incident/views.py

```diff
@@ -154,6 +158,26 @@ def update_incident(
     return incident
 
 
+@router.post("/{incident_id}/join", summary="Join an incident.")
+def join_incident(
+    *,
+    db_session: Session = Depends(get_db),
+    incident_id: str,
```

Comment:
nit: shouldn't `incident_id` be of type `int`? Same goes for the one in `service.get()`